docs/.vuepress/components/Search.vue

```diff
@@ -52,67 +58,99 @@
 </template>
 
 <script>
+import debounce from "tiny-debounce";
+import nprogress from "nprogress";
+
 export default {
   props: {
     threshold: {
       type: Number,
       default: 300
     }
   },
-  data () {
+
+  data() {
     return {
       scrollTop: null,
       buckets: [],
       apps: [],
-      query: '',
-      apps_found: false,
-    }
+      query: "",
+      apps_found: false
+    };
   },
-  mounted () {
-    this.axios.get('https://api.github.com/search/repositories?q=topic:scoop-bucket').then((response) => {
-      this.buckets = response.data.items
-    })
+  mounted() {
+    nprogress.configure({ showSpinner: false });
+    nprogress.start();
+    this.axios
+      .get("https://api.github.com/search/repositories?q=topic:scoop-bucket")
+      .then(response => {
+        this.buckets = response.data.items;
+        nprogress.done();
+      });
   },
   methods: {
-    search (value) {
-      let buckets_string = ''
-      this.buckets.forEach((bucket) => {
-        buckets_string += `+repo:${bucket.full_name}`
-      })
-      this.axios.get(`https://api.github.com/search/code?q=${this.query}+in:file+extension:json${buckets_string}`).then((response) => {
-        this.apps = response.data.items
-        this.apps_found = response.data.total_count > 0
-      })
-    },
-    scrollToTop () {
-      window.scrollTo({ top: 0, behavior: 'smooth' })
-      this.scrollTop = 0
+    search: debounce(function() {
+
+      // One char always return zero
+      if (this.query.length < 2) return
+
+      nprogress.start();
+
+      let buckets = "";
+      this.buckets.forEach(bucket => {
+        buckets += `+repo:${bucket.full_name}`;
+      });
+
+      buckets = encodeURI(buckets);
+      const query = encodeURI(this.query);
+      const url = `https://api.github.com/search/code?q=${query}+in:file+extension:json${buckets}`
+      this.axios
+        .get(url)
+        .catch(err => {
+          nprogress.done();
+        })
+        .then(({ data: { items, total_count } }) => {
+          this.apps = items;
+          console.log({ total_count });
+          this.apps_found = total_count > 0;
+          nprogress.done();
+        });
+    }, 1000),
+
+    scrollToTop() {
+      window.scrollTo({ top: 0, behavior: "smooth" });
+      this.scrollTop = 0;
     }
   },
   computed: {
-    show () {
-      return this.scrollTop > this.threshold
+    show() {
+      return this.scrollTop > this.threshold;
     }
   }
-}
+};
 </script>
 
 <style lang="stylus">
-$borderColor = #eaecef
+$borderColor = #eaecef;
```

Comment:
I used `lang="stylus"` so `;`, `{` and `}` are not needed in the style

Reply:
Maybe we better adding lint to the workflow so it formats automatically